### PR TITLE
[qt6][spatialite] QRegEx -> QRegularExpression

### PR DESCRIPTION
**Pull request by nirvn**
## Description

Another one.

### CODE REVIEW

File: src/providers/spatialite/qgsspatialiteprovider.cpp [hunk 1]
```diff
@@ -5045,7 +5045,7 @@ bool QgsSpatiaLiteProvider::checkLayerType()
         // Check if the whole sql is aliased i.e. '(SELECT * FROM \\"somedata\\" as my_alias\n)'
         if ( tableAlias.isEmpty() )
         {
-          regex.setPattern( QStringLiteral( R"re(\s+AS\s+(\w+)\n?\)?$)re" ) );
+          re.setPattern( QStringLiteral( R"re(\s+AS\s+(\w+)\n?\)?$)re" ) );
```
Comment (roya0045) [inline src/providers/spatialite/qgsspatialiteprovider.cpp:5048]: This seems to be a mistake, the variable regex is unchanged on 4999.

Comment (nirvn) [reply src/providers/spatialite/qgsspatialiteprovider.cpp:5048]: Na this is actually correcting a mistake

File: src/providers/spatialite/qgsspatialiteprovider.cpp [hunk 2]
```diff
@@ -4990,13 +4990,13 @@ bool QgsSpatiaLiteProvider::checkLayerType()
     // get a new alias for the subquery
     int index = 0;
     QString alias;
-    QRegExp regex;
+    QRegularExpression regex;
     do
     {
       alias = QStringLiteral( "subQuery_%1" ).arg( QString::number( index++ ) );
-      QString pattern = QStringLiteral( "(\\\"?)%1\\1" ).arg( QRegExp::escape( alias ) );
+      QString pattern = QStringLiteral( "(\\\"?)%1\\1" ).arg( QRegularExpression::escape( alias ) );
```
Comment (roya0045) [inline src/providers/spatialite/qgsspatialiteprovider.cpp:4997]: escape will need ifdef as its a 5.15 thing.

Comment (nirvn) [reply src/providers/spatialite/qgsspatialiteprovider.cpp:4997]: Not according to Qt's documentation and the green CI tests which compile under pre 5.15.

Comment (roya0045) [reply src/providers/spatialite/qgsspatialiteprovider.cpp:4997]: Yeah maybe only the QStringView is 5.15, qt doc is unclear if this was a base thing.

Comment (nyalldawson) [reply src/providers/spatialite/qgsspatialiteprovider.cpp:4997]: Ok confirmed -- it's in 5.11 https://doc.qt.io/archives/qt-5.11/qregularexpression.html#escape

I guess we can cleanup the ifdefs we've used elsewhere for this one then!

**Comment by nirvn** [comment]
@3nids , is it normal that cppcheck_16_04 still shows up (I assume it'll idle forever)?

image

**Comment by 3nids** [comment]
I updated the branch protection rule, should be good now

**Pull request merged**